lib/suppressions.cpp

```diff
@@ -112,6 +112,92 @@ std::string Suppressions::parseXmlFile(const char *filename)
     return "";
 }
 
+std::vector<Suppressions::Suppression> Suppressions::parseMultiSuppressComment(std::string comment, std::string *errorMessage)
+{
+    std::vector<Suppression> suppressions;
+    if (comment.size() < 2)
+        return suppressions;
+
+    size_t suppress_position = comment.find("cppcheck-suppress");
+    if (suppress_position == std::string::npos)
```

Comment:
I believe that no text should be allowed before "cppcheck-suppress". I suggest adding a check somewhere:

    suppress_position == comment.find_first_not_of(" ")